lib/db/mongodb/plugins/orderable.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+module.exports = (schema, options) => {
+  const { fieldName }       = options
+  const orderableField      = {}
+  orderableField[fieldName] = { type: Number }
+
+  const orderableIndex      = { _deleted: 1 }
+  orderableIndex[fieldName] = 1
+
+  schema.add(orderableField)
+
+  const sortBy = {}
+  sortBy[fieldName] = -1
+  schema.pre('save', function (next) {
+    this.constructor.find({}).sort(sortBy).limit(1)
```

Comment:
Are you sure index is used in this case, do we need to add `_delete: true` or it's added automatically?